frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-install.tsx

```diff
@@ -0,0 +1,384 @@
+      if (index < newSelectedRowData.length) {
+        // preserve existing selection state
+        newSelectedRowData[index] = {
+          ...row,
+          uid: row.metadata.uid,
+          selected: newSelectedRowData[index].selected,
+        };
+      } else {
+        // set initial selection state from storage here if necessary...for now, initialize it false
+        newSelectedRowData.push({ ...row, uid: row.metadata.uid, selected: false });
+      }
+    });
+
+    if (virtualRowIndex !== -1) {
+      // set the selection based on virtualRowIndex node, it should exist in the array now
+      newSelectedRowData[virtualRowIndex].selected = isSelected;
+    } else {
+      // selectAll
+      newSelectedRowData = _.map(newSelectedRowData, (row) => ({ ...row, selected: isSelected }));
+    }
+    setSelectedNodeData(newSelectedRowData);
+
+    setSelectedNodesCnt(_.filter(newSelectedRowData, 'selected').length);
+  };
+
+  React.useEffect(() => {
+    selectedNodesCnt >= minSelectedNode ? setCreateBtnDisabled(false) : setCreateBtnDisabled(true);
+  }, [selectedNodesCnt]);
+
+  const handleRequestSizeInputChange = (obj) => {
+    setRequestSizeUnit(obj.unit);
+    setRequestSizeValue(obj.value);
+  };
+
+  const handleStorageClass = (sc) => {
+    setStorageClass(_.get(sc, 'metadata.name'));
+  };
+
+  const OCSNodeList = (nodeProps) => (
+    <NodeList
+      {...nodeProps}
+      data={nodeProps.data}
+      customData={selectedNodeData}
+      onSelect={(event, isSelected, virtualRowIndex) =>
+        onSelect(event, isSelected, virtualRowIndex, nodeProps.data)
+      }
+    />
+  );
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setProgress(true);
+    setError('');
+
+    const labelPath = '/metadata/labels';
+    const promises = [];
+
+    // labeling the selected nodes
+    const selectedData = _.filter(selectedNodeData, 'selected');
+    promises.push(
+      ...selectedData.map(function(node) {
+        const labels = SelectorInput.arrayify(_.get(node, labelPath.split('/').slice(1)));
+        const patch = [
+          {
+            op: labels.length ? 'replace' : 'add',
+            value: SelectorInput.objectify(labels),
+            path: labelPath,
+          },
+        ];
+        patch[0].value = { ...patch[0].value, 'cluster.ocs.openshift.io/openshift-storage': '' };
+        return k8sPatch(NodeModel, node, patch);
+      }),
+    );
+
+    // tainting the selected nodes
+    promises.push(
+      [...selectedData]
+        .filter((node) => {
+          const roles = getNodeRoles(node);
+          // don't taint master nodes as its already tainted
+          return roles.indexOf('master') === -1;
+        })
+        .map((node) => {
+          const taints = node.spec.taints
+            ? [
+                ...node.spec.taints,
+                { key: 'node.ocs.openshift.io/storage', value: 'true', effect: 'NoSchedule' },
```

Comment:
Make `'node.ocs.openshift.io/storage'`  as `const ocsTaint` or something? Prevents repetition and helps making changes easier.